account_general_ledger_report/models/general_ledger_report.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+import datetime
+from openerp import models, api, fields, _
+
+
+class AccountGeneralLedgerReport(models.Model):
+    _name = 'account.general.ledger.report'
+
+    name = fields.Char(
+        string='Name',
+        readonly=True,
+    )
+    fiscalyear_id = fields.Many2one(
+        'account.fiscalyear',
+        string='Fiscal Year',
+        readonly=True,
+    )
+    target_move = fields.Selection(
+        [('posted', 'All Posted Entries'),
+         ('all', 'All Entries')],
+        string='Target Moves',
+        readonly=True,
+    )
+    reconcile_cond = fields.Selection(
+        [('all', 'All Items'),
+         ('open_item', 'Open Items'),
+         ('reconciled', 'Full Reconciled')],
+        string=" ",
+        readonly=True,
+    )
+    account_ids = fields.Many2many(
+        'account.account',
+        string="Accounts",
+        required=True,
+    )
+    amount_currency = fields.Boolean(
+        string='With Currency',
+        default=True,
+    )
+    line_ids = fields.One2many(
+        'account.general.ledger.line',
+        'report_id',
+        string='Details'
+    )
+
+    @api.model
+    def _get_moves(self, fiscalyear_id, target_move, reconcile_cond,
+                   account_ids, amount_currency):
+        Period = self.env['account.period']
+        Move = self.env['account.move.line']
+        Fiscalyear = self.env['account.fiscalyear'].browse(fiscalyear_id)
```

Comment:
Do not use Capital.
Normally, all variable should be in lower case.
Only Object declared will be in first Capital.